internal/storage/file_storage.go

```diff
@@ -0,0 +1,198 @@
+package storage
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"go.uber.org/zap"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type FileStorage struct {
+	fileName string
+	Log      *zap.Logger
+}
+
+var _ Repository = (*FileStorage)(nil)
+
+func NewFileStorage(fileName string, logger *zap.Logger) *FileStorage {
+	err := os.MkdirAll(filepath.Dir(fileName), 0774)
+	if err != nil {
+		logger.Error("Ошибка создания файла")
+	}
+	logger.Debug("Создали дирректорию", zap.String("dir", filepath.Dir(fileName)))
+	return &FileStorage{
+		fileName: fileName,
+		Log:      logger,
+	}
+}
+
+func (f *FileStorage) AddURL(data URLData) error {
+	file, err := os.OpenFile(f.fileName, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0774)
+	defer func() {
+		err := file.Close()
+		if err != nil {
+			f.Log.Error("Ошибка закрытия файла", zap.Error(err))
+		}
+	}()
+	if err != nil {
+		return err
+	}
+	encoder := json.NewEncoder(file)
+	err = encoder.Encode(data)
+	if err != nil {
+		return err
+	}
+	err = file.Close()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (f *FileStorage) CheckID(id string) (URLData, bool, error) {
+	file, err := os.OpenFile(f.fileName, os.O_RDONLY|os.O_CREATE|os.O_APPEND, 0774)
+	defer func() {
+		err := file.Close()
+		if err != nil {
+			f.Log.Error("Ошибка закрытия файла", zap.Error(err))
+		}
+	}()
+
+	if err != nil {
+		return URLData{}, false, err
+	}
+	scanner := bufio.NewScanner(file)
+	var shortUrls []URLData
+
+	for scanner.Scan() {
```

Comment:
CHANGE REQUEST: please provide me asymptotic estimation about this method.

Reply:
Optimised